runelite-client/src/main/java/net/runelite/client/plugins/wasdcamera/WASDCameraPlugin.java

```diff
@@ -181,9 +182,24 @@ void unlockChat()
 				{
 					final boolean isChatboxTransparent = client.isResized() && client.getVar(Varbits.TRANSPARENT_CHATBOX) == 1;
 					final Color textColor = isChatboxTransparent ? JagexColors.CHAT_TYPED_TEXT_TRANSPARENT_BACKGROUND : JagexColors.CHAT_TYPED_TEXT_OPAQUE_BACKGROUND;
-					chatboxInput.setText(client.getLocalPlayer().getName() + ": " + ColorUtil.wrapWithColorTag(client.getVar(VarClientStr.CHATBOX_TYPED_TEXT) + "*", textColor));
+					chatboxInput.setText(getPlayerNameWithIcon() + ": " + ColorUtil.wrapWithColorTag(client.getVar(VarClientStr.CHATBOX_TYPED_TEXT) + "*", textColor));
 				}
 			}
 		}
 	}
+
+	private String getPlayerNameWithIcon()
+	{
+		switch (client.getAccountType())
```

Comment:
Can this be changed to determine the prefix for the player name? e.g.

```
String playerPrefix = '';
switch (client.getAccountType())
{
    case IRONMAN:
        playerPrefix = IconID.IRONMAN;
        break;
    case ULTIMATE_IRONMAN:
        playerPrefix = IconID.ULTIMATE_IRONMAN;
        break;
    case HARDCORE_IRONMAN:
        playerPrefix = IconID.HARDCORE_IRONMAN;
        break;
}
return playerPrefix + client.getLocalPlayer().getName();
```

That way, refactoring the way the prefix is gotten, or refactoring the player name call would require less modification to this method.

Reply:
I did this before, and DB didn't want me using `toString` in here, to just let the jvm do it. Removing the `toString` calls causes the compiler to fail, the types are incompatible and it doesn't try to `toString` unless told to.

Reply:
Hm... I suppose it's fine as-is then.

Reply:
No, you didnt @Hydrox6 . Just save the playerPrefix as enum and at the end you can do playerPredix + getName and it will work. And that is better that what you have now with duplicated code